cmd/firectl/main.go

```diff
@@ -188,23 +196,49 @@ func main() {
 		log.Fatalf("Invalid vsock specification: %s", err)
 	}
 
+	var fifo *os.File
+	if len(opts.FcFifoLogFile) != 0 {
+		dir, err := ioutil.TempDir(os.TempDir(), "fcfifo")
+		if err != nil {
+			log.Fatalf("Fail to create temporary directory: %v", err)
+		}
+
+		defer os.RemoveAll(dir)
```

Comment:
FIFO management still feels a bit off. Combinations of certain firectl flags result in weird behavior. For example:

(most flags left out, only log/fifo related flags are shown)


```
$ firectl --firecracker-log=/tmp/y
...
INFO[0000] Configured VMM logging to /tmp/fcfifo558137878/fc_fifo, metrics to /tmp/fcfifo558137878/fc_metrics_fifo: [PUT /logger][204] putLoggerNoContent  
```
The metrics fifo sticks around for the lifetime of the process. Since I never asked for a metrics FIFO, this is surprising to me.

```
$ firectl --firecracker-log=/tmp/log --vmm-log-fifo=/tmp/fc.log --metrics-fifo=/tmp/metrics
...
ERRO[0000] Unable to set up logging: file exists        
WARN[0000] setupLogging() returned file exists. Continuing anyway. 
```
I asked for all 3, and got an unhelpful error.
```
$ firectl --metrics-fifo=/tmp/metrics
...
INFO[0000] VMM logging and metrics disabled.            
```
I asked for only one FIFO, but got nothing at all. Since firecracker wants both FIFOs configured, I think we should create the unspecified one in a tempdir and immediately unlink it. (I realize that this is behavior predates this PR and is largely driven by Firecracker's behavior, but we should handle it better.)

Reply:
> The metrics fifo sticks around for the lifetime of the process. Since I never asked for a metrics FIFO, this is surprising to me.

I thought this was the simplest approach, since it does delete it, just until the process ends. Otherwise, we wouldn't be able to delete until we've opened the file. However, this does open the question for a potential change in the design. For instance, if we did something similar to the `FifoLogWriter` being an `io.Writer` with the log and metric fifo, we'd be able to handle that at the `firectl` level. I also think this is the right approach. What do you think?

> I asked for all 3, and got an unhelpful error.

Fixed and was a issue with how interfaces deal with `nil` values.

> I asked for only one FIFO, but got nothing at all. Since firecracker wants both FIFOs configured, I think we should create the unspecified one in a tempdir and immediately unlink it. (I realize that this is behavior predates this PR and is largely driven by Firecracker's behavior, but we should handle it better.)

Fair enough. I'll update. If we go with the design above, I can reuse some of the logic in `firectl`